### ISSUE TITLE
Negative sampling

### ISSUE THREAD

**Issue by georgesms** [matwerner/word-embedding#3]
Implement negative sampling as an alternative for softmax. IDEA: generate random integers and map through theano the backpropagation only regrading these.

"The paper says that selecting 5-20 words works well for smaller datasets, and you can get away with only 2-5 words for large datasets."

http://mccormickml.com/2017/01/11/word2vec-tutorial-part-2-negative-sampling/
